File: Comms Protocol Java/Comms Protocol Java/comms/protocol/java/AndroidSensor.java
package comms.protocol.java;

import java.nio.ByteBuffer;

public class AndroidSensor {
	private float[] values;
	private long timeStamp;

	public static int NUM_BYTES_PER_LONG = 8;
	public static int NUM_BYTES_PER_FLOAT = 4;

	public AndroidSensor(int numValues) {
		values = new float[numValues];
		timeStamp = 0;
	}

	public AndroidSensor(float[] vals, long tStamp) {
		values = vals;
		timeStamp = tStamp;
	}

	public void SetValues(float[] vals) {
		values = vals;
	}

	public void SetTimeStamp(long tStamp) {
		timeStamp = tStamp;
	}

	public long GetTimeStamp() {
		return timeStamp;
	}

	public float[] GetValues() {
		return values;
	}

	public int Length() {
		return values.length;
	}

	public byte[] GetBytes() {
		ByteBuffer buffer = ByteBuffer.allocate((values.length * NUM_BYTES_PER_FLOAT) + NUM_BYTES_PER_LONG);
		for (int i = 0; i < values.length; i++)
			buffer.putFloat(values[i]);
		buffer.putLong(timeStamp);
		return buffer.array();
	}

	public void SetBytes(byte[] bytes) {
		if (bytes.length < ((values.length * NUM_BYTES_PER_FLOAT) + NUM_BYTES_PER_LONG))
			return;
		ByteBuffer buffer = ByteBuffer.wrap(bytes);
		for (int i = 0; i < values.length; i++)
			values[i] = buffer.getFloat();
		timeStamp = buffer.getLong();
	}
}

File: Comms Protocol CSharp/Comms Protocol CSharp/AndroidSensor.cs
using System;

namespace Comms_Protocol_CSharp
{
    public class AndroidSensor
    {
        private float[] _values;
        private long _timeStamp;

        public static int NUM_BYTES_PER_LONG = 8;
        public static int NUM_BYTES_PER_FLOAT = 4;

        public AndroidSensor(int numValues)
        {
            _values = new float[numValues];
            _timeStamp = 0;
        }

        public AndroidSensor(float[] vals, long tStamp)
        {
            _values = vals;
            _timeStamp = tStamp;
        }

        public void SetValues(float[] vals)
        {
            _values = vals;
        }

        public void SetTimeStamp(long tStamp)
        {
            _timeStamp = tStamp;
        }

        public long GetTimeStamp()
        {
            return _timeStamp;
        }

        public float[] GetValues()
        {
            return _values;
        }

        public int Length()
        {
            return _values.Length;
        }

        public byte[] GetBytes()
        {
            int numBytesInVal = _values.Length * NUM_BYTES_PER_FLOAT;
            byte[] rtn = new byte[numBytesInVal + NUM_BYTES_PER_LONG];
            Buffer.BlockCopy(_values, 0, rtn, 0, numBytesInVal);
            long[] tmp = new long[] { _timeStamp };
            Buffer.BlockCopy(tmp, 0, rtn, numBytesInVal, NUM_BYTES_PER_LONG);
            return rtn;
        }

        public void SetBytes(byte[] bytes)
        {
            if (bytes.Length < ((_values.Length * NUM_BYTES_PER_FLOAT) + NUM_BYTES_PER_LONG))
                return;
            Buffer.BlockCopy(bytes, 0, _values, 0, (_values.Length * NUM_BYTES_PER_FLOAT));
            long[] tmp = new long[1];
            Buffer.BlockCopy(bytes, (_values.Length * NUM_BYTES_PER_FLOAT), tmp, 0, NUM_BYTES_PER_LONG);
            _timeStamp = tmp[0];
        }
    }
}
